### ISSUE TITLE
BLT Validate YAML fails due to yaml-cli

### ISSUE THREAD

**Issue by billdaff** [acquia/blt#4390]
Sorry if this is not the right place, as it may be a bug in grasmash/yaml-cli, or with my own environment (DDEV). I have verified that my config/yaml file is correct. I am on blt version 13.0.1.0, ddev version 1.17.6, php 7.4.22 and when it runs the yaml linter, it fails with the following error:

<pre> Iterating over fileset files.yaml...
''/var/www/html/vendor/bin/yaml-cli' lint '/var/www/html/config/default/views.view.orders_and_decisions.yml'' exited with code 255 
[error]  Executing `'/var/www/html/vendor/bin/yaml-cli' lint '%s'` against fileset(s) files.yaml returned a non-zero exit code.`
For troubleshooting guidance and support, see https://docs.acquia.com/blt/support/ 
The command failed. This often indicates a problem with your configuration. Review the command output above for more detailed errors, and consider re-running with verbose output for more information.
[CompletionWrapper]  Command `validate:yaml:lint:files 'config/default/views.view.orders_and_decisions.yml'` exited with code 1.
For troubleshooting guidance and support, see https://docs.acquia.com/blt/support/ 
[CompletionWrapper]  Exit code 1 </pre>

and when I run the top command I get the following error:
<pre>PHP Warning:  require(application.php): failed to open stream: No such file or directory in /var/www/html/vendor/bin/yaml-cli(20) : eval()'d code on line 2
PHP Stack trace:
PHP   1. {main}() /var/www/html/vendor/bin/yaml-cli:0
PHP   2. eval() /var/www/html/vendor/bin/yaml-cli:20

Warning: require(application.php): failed to open stream: No such file or directory in /var/www/html/vendor/bin/yaml-cli(20) : eval()'d code on line 2

Call Stack:
    0.2041     360560   1. {main}() /var/www/html/vendor/bin/yaml-cli:0
    0.2043     361736   2. eval('
require 'application.php';') /var/www/html/vendor/bin/yaml-cli:20

PHP Fatal error:  require(): Failed opening required 'application.php' (include_path='.:/usr/share/php') in /var/www/html/vendor/bin/yaml-cli(20) : eval()'d code on line 2
PHP Stack trace:
PHP   1. {main}() /var/www/html/vendor/bin/yaml-cli:0
PHP   2. eval() /var/www/html/vendor/bin/yaml-cli:20

Fatal error: require(): Failed opening required 'application.php' (include_path='.:/usr/share/php') in /var/www/html/vendor/bin/yaml-cli(20) : eval()'d code on line 2

Call Stack:
    0.2041     360560   1. {main}() /var/www/html/vendor/bin/yaml-cli:0
    0.2043     361736   2. eval('
require 'application.php';') /var/www/html/vendor/bin/yaml-cli:20 </pre>

I noticed the same error happens when just running yaml-cli as well. In my troubleshooting I figured the call to require application.php didn't have the relative path set correctly. If I go into `/var/www/html/vendor/grasmash/yaml-cli/bin/yaml-cli` and change the require path to `require '/var/www/html/vendor/grasmash/yaml-cli/bin/application.php';` then the blt validate and git commit hooks pass as expected.

I am not sure how to correctly fix this, or even where honestly. This was just my attempt at getting it to work and finding what the issue may be. Please advise when you get a chance.

**Comment by mikemadison13** [comment]
my very rough read of this is either that yaml-cli didn't install correctly (might be worth rm -rf vendor and composer install) OR there's an issue with yaml-cli itself. but that's just a guess!

**Comment by billdaff** [comment]
Just tried the rm -rf, didnt work. But I did change my editted require to `require __DIR__ . '/application.php';` in `/var/www/html/vendor/grasmash/yaml-cli/bin/yaml-cli` and it worked as well. So I went back into `/var/www/html/vendor/bin/yaml-cli` and I think the issue is that this file is now being automatically generated by composer because it is not what they have in their repo https://github.com/grasmash/yaml-cli/blob/master/bin/yaml-cli The following comment was also added to the top of the newly generated file:
<pre>
/**
 * Proxy PHP file generated by Composer
 *
 * This file includes the referenced bin path (../grasmash/yaml-cli/bin/yaml-cli) using eval to remove the shebang if present
 *
 * @generated
 */
</pre>


Does the blt composer.json need to lock in a specific version or something? I hijacked a coworkers computer and did an rm -rf in his vendor folder and did a composer install and it changed his file and it regenerated the file to what I had, which broke it for him to with same issues. I am not sure why composer is regenerating this file (I noticed other vendor bin files are being regenerated as well though). I'll continue looking into it and post any updates I find.
